### PR TITLE
gh-124102: Update internal PCbuild docs and correct WiX registry

### PR DESCRIPTION
**Pull request by Wulian233**
skip news
@zooba I got something wrong in the previous PR

- Corrected the incorrect information in the previous PR

- Fixed the incorrect WiX 3.14 registry.

* Issue: gh-124102

### CODE REVIEW

**Comment by Wulian233** [comment]
This pr should not be backported, there will be conflicts. Another document PR that requires a backport is a necessary part

File: Tools/msi/wix.props [hunk 1]
```diff
@@ -5,8 +5,7 @@
   <PropertyGroup>
     <WixInstallPath Condition="'$(WixInstallPath)' == '' and Exists('$(MSBuildThisFileDirectory)\Wix')">$(MSBuildThisFileDirectory)\Wix\</WixInstallPath>
     <WixInstallPath Condition="'$(WixInstallPath)' == '' and Exists('$(ExternalsDir)\windows-installer\wix-314')">$(ExternalsDir)\windows-installer\wix-314\</WixInstallPath>
-    <WixInstallPath Condition="'$(WixInstallPath)' == ''">$(Registry:HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows Installer XML\3.10@InstallRoot)</WixInstallPath>
-    <WixInstallPath Condition="'$(WixInstallPath)' == ''">$(Registry:HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Windows Installer XML\3.10@InstallRoot)</WixInstallPath>
+    <WixInstallPath Condition="'$(WixInstallPath)' == ''">$(Registry:HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Windows Installer XML\3.14@InstallRoot)</WixInstallPath>
```
Comment (zooba) [inline Tools/msi/wix.props]: Let's just delete all the registry checks and add a note to the readme along the lines of "if you are not using the copy of Wix installed by `Tools\msi\get_externals.bat`, then you will need to set the `WixInstallPath` environment variable before building."

Comment (Wulian233) [reply Tools/msi/wix.props]: deleted :)

File: Doc/using/windows.rst [hunk 2]
```diff
@@ -25,7 +25,7 @@ As specified in :pep:`11`, a Python release only supports a Windows platform
 while Microsoft considers the platform under extended support. This means that
 Python |version| supports Windows 10 and newer. If you require Windows 7
 support, please install Python 3.8. If you require Windows 8.1 support,
-please install Python 3.12.
+please install Python 3.13.
```
Comment (zooba) [inline Doc/using/windows.rst]: Responded to this on a different PR - according to [PEP 11](https://peps.python.org/pep-0011/), 3.13 doesn't support Windows 8.1, so we don't want to recommend it here. It works today (we believe, I don't think anyone is testing it), but we reserve the right to break it if necessary.

**Comment by Wulian233** [comment]
@zooba Could you please review this again? I have already resolved the issues mentioned :)

**Pull request merged**